aten/src/ATen/native/cuda/MultinomialKernel.cu

```diff
@@ -60,10 +64,10 @@ void renormRows(Tensor& t) {
   auto props = at::cuda::getCurrentDeviceProperties();
   CUDA_KERNEL_ASSERT(props != NULL);
   int numSM = props->multiProcessorCount;
-  int maxThreads = props->maxThreadsPerBlock;
+  int64_t maxThreads = props->maxThreadsPerBlock;
 
   dim3 grid(rows < numSM * 4 ? rows : numSM * 4);
-  dim3 block(cols < maxThreads ? cols : maxThreads);
+  dim3 block(std::min(maxThreads, C10_WARP_SIZE * div_up(cols, C10_WARP_SIZE)));
```

Comment:
Rounding up to a multiple of `C10_WARP_SIZE` is okay because the kernel structure is:
```cpp
    for (int64_t col = threadIdx.x; col < cols; col += blockDim.x) {
      // local sum
    }

    sum = cuda_utils::BlockReduceSum(sum, smem);
    // broadcast block sum to all threads...

    for (int64_t col = threadIdx.x; col < cols; col += blockDim.x) {
        // use block sum
    }
```